test_fms/parser/test_output_yaml.F90

```diff
@@ -138,11 +138,11 @@ program test_output_yaml
 call fms_f2c_string (v1(1)%val1,"time to eat")
 call fms_f2c_string (k1(1)%key2,"location")
 call fms_f2c_string (v1(1)%val2,"Bridgewater, NJ")
-call yaml_out_add_level2key( "order", k1(1))
+call yaml_out_add_level2key( 5, "order", k1(1))
```

Comment:
This works.  I think you should make at least one of these use the `len` function and maybe even a `trim`.  That's more likely how it will be used in practice.  I don't think you need to change them all, just one.
```fortran
order="order"
call yaml_out_add_level2key( len(trim(order)), order, k1(1))
```

Reply:
I'll just add a wrapper for it in fortran, that should make it so that you don't have to worry about the length.

btw theres a len_trim() fortran intrinsic, does both at once